drivers/adxl362/adxl362.c

```diff
@@ -0,0 +1,652 @@
+#define ADXL362_STATUS_FIFO_OVERRUN_MASK   (0x08)
+#define ADXL362_STATUS_ACT_MASK            (0x10)
+#define ADXL362_STATUS_INACT_MASK          (0x20)
+#define ADXL362_STATUS_AWAKE_MASK          (0x40)
+#define ADXL362_STATUS_ERR_USER_REGS_MASK  (0x80)
+
+#define ADXL362_FILTER_RANGE_SHIFT         (6)
+#define ADXL362_FILTER_RANGE_MASK          (0x03 << ADXL362_FILTER_RANGE_SHIFT)
+#define ADXL362_FILTER_HALF_BW_SHIFT       (4)
+#define ADXL362_FILTER_HALF_BW_MASK        (0x01 << ADXL362_FILTER_HALF_BW_SHIFT)
+#define ADXL362_FILTER_ODR_SHIFT           (0)
+#define ADXL362_FILTER_ODR_MASK            (0x07 << ADXL362_FILTER_ODR_SHIFT)
+
+#define ADXL362_FIFO_CONTROL_TEMP_SHIFT    (2)
+#define ADXL362_FIFO_CONTROL_TEMP_MASK     (0x01 << ADXL362_FILTER_RANGE_SHIFT)
+#define ADXL362_FIFO_CONTROL_AH_SHIFT      (3)
+#define ADXL362_FIFO_CONTROL_AH_MASK       (0x01 << ADXL362_FILTER_HALF_BW_SHIFT)
+#define ADXL362_FIFO_CONTROL_MODE_SHIFT    (0)
+#define ADXL362_FIFO_CONTROL_MODE_MASK     (0x03 << ADXL362_FILTER_ODR_SHIFT)
+
+#define ADXL362_SELF_TEST_MASK             (0x01)
+/* Time to wait during self test until values have stabilized */
+#define ADXL362_SELF_TEST_SLEEP            (100 * MS_IN_USEC)
+/* Self test expected values */
+#define ADXL362_SELF_TEST_XOFF_MIN         (450)
+#define ADXL362_SELF_TEST_XOFF_MAX         (710)
+#define ADXL362_SELF_TEST_YOFF_MIN         (-710)
+#define ADXL362_SELF_TEST_YOFF_MAX         (-450)
+#define ADXL362_SELF_TEST_ZOFF_MIN         (350)
+#define ADXL362_SELF_TEST_ZOFF_MAX         (650)
+/* The self-test offset is dependent on the supply voltage, see data sheet:
+ * Table 22. Self Test Output Scale Factors for Different Supply Voltages, Vs */
+/* represented as a power of two fraction (FACTOR / 2**SHIFT) */
+/* Choose this shift so that the 32 bit intermediate variable does not overflow,
+ * 16 bits should be safe for the whole sensor range and still close to optimum */
+#define ADXL362_SELF_TEST_VSS_SCALE_SHIFT  (16)    /* denominator exponent */
+/* The values below were interpolated from the given table for a 3.3 V supply
+ * voltage (2.95x is the resulting factor) */
+/* The division by 100 is placed here with only numeric constants to allow the
+ * compiler to optimize away the division completely */
+#define ADXL362_SELF_TEST_VSS_SCALE_FACTOR ((295UL << ADXL362_SELF_TEST_VSS_SCALE_SHIFT) / 100)
+/* convenience macro */
+#define ADXL362_SELF_TEST_RESCALE_LIMIT(x) \
+    (((x) * ADXL362_SELF_TEST_VSS_SCALE_FACTOR) >> ADXL362_SELF_TEST_VSS_SCALE_SHIFT)
+
+/**
+ * @internal
+ * @brief Read registers from the device
+ *
+ * @param[in]  dev   pointer to device descriptor
+ * @param[in]  addr  register address
+ * @param[out] buf   output buffer
+ * @param[in]  count number of bytes to read
+ *
+ * @return 0 on success
+ * @return <0 on error
+ */
+static int adxl362_read_regs(const adxl362_t *dev, uint8_t addr, uint8_t *buf, size_t count)
+{
+    DEBUG("adxl362_read_regs: %p, 0x%02x, %p, %lu\n", (void *)dev,
```

Comment:
You can replace the function name `adxl362_read_regs` in the format string with `%s` and give `, __FUNCTION__` as a parameter to `DEBUG`.